docs/source/python/orc.rst

```diff
@@ -0,0 +1,261 @@
+
+:func:`~pyarrow.orc.write_table()` has a number of options to
+control various settings when writing an ORC file.
+
+* ``file_version``, the ORC format version to use.  ``'0.11'`` ensures
+  compatibility with older readers, while ``'0.12'`` is the newer one.
+* ``stripe_size``, to control the approximate size of data within a column 
+  stripe. This currently defaults to 64MB.
+
+See the :func:`~pyarrow.orc.write_table()` docstring for more details.
+
+There are some additional data type handling-specific options
+described below.
+
+Omitting the DataFrame index
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When using ``pa.Table.from_pandas`` to convert to an Arrow table, by default
+one or more special columns are added to keep track of the index (row
+labels). Storing the index takes extra space, so if your index is not valuable,
+you may choose to omit it by passing ``preserve_index=False``
+
+.. ipython:: python
+
+   df = pd.DataFrame({'one': [-1, np.nan, 2.5],
+                      'two': ['foo', 'bar', 'baz'],
+                      'three': [True, False, True]},
+                      index=list('abc'))
+   df
+   table = pa.Table.from_pandas(df, preserve_index=False)
+
+Then we have:
+
+.. ipython:: python
+
+   po.write_table(table, 'example_noindex.orc')
+   t = po.read_table('example_noindex.orc')
+   t.to_pandas()
+
+Here you see the index did not survive the round trip.
+
+Finer-grained Reading and Writing
+---------------------------------
+
+``read_table`` uses the :class:`~.ORCFile` class, which has other features:
+
+.. ipython:: python
+
+   orc_file = po.ORCFile('example.orc')
+   orc_file.metadata
+   orc_file.schema
+   orc_file.nrows
+
+See the :class:`~pyarrow.orc.ORCFile()` docstring for more details.
+
+As you can learn more in the `Apache ORC format
+<https://orc.apache.org/specification/>`_, an ORC file consists of
+multiple stripes. ``read_table`` will read all of the stripes and
+concatenate them into a single table. You can read individual stripes with
+``read_stripe``:
+
+.. ipython:: python
+
+   orc_file.nstripes
+   orc_file.read_stripe(0)
+
+We can write an ORC file using ``ORCWriter``:
+
+.. ipython:: python
+
+   with po.ORCWriter('example2.orc') as writer:
+      writer.write_table(table)
+
+Compression
+---------------------------------------------
+
+The data pages within a column in a row group can be compressed after the
+encoding passes (dictionary, RLE encoding). In PyArrow we don't use compression
+by default, but Snappy, ZSTD, Gzip/Zlib, and LZ4 are also supported:
+
+.. code-block:: python
+
+   po.write_table(table, where, compression='uncompressed')
+   po.write_table(table, where, compression='gzip')
+   po.write_table(table, where, compression='zstd')
+   po.write_table(table, where, compression='snappy')
+
+Snappy generally results in better performance, while Gzip may yield smaller
+files.
+
+Partitioned Datasets (Multiple Files)
+------------------------------------------------
```

Comment:
This section doesn't showcase any example or API, is it useful to have it?

Reply:
I agree that it is not that useful now. I removed it. When we complete the ORC dataset feature we will add it.